bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -797,6 +763,10 @@ public void openComplete(int newrc, final LedgerHandle newlh, Object newctx) {
                     return;
                 }
 
+                if (dryrun) {
+                    System.out.println("Recovered ledger " + lId + " : " + (fenceRequired ? "[fence required]" : ""));
```

Comment:
this was made to use `System.out` for dryrun in bk shell. I have a upcoming change in bk shell to use `dryrun`.

Reply:
Doesn't BookKeeper shell use log4j? If not, we should make it do so.

Reply:
@ivankelly it uses log4j. the dryrun is printing on the console rather than using log4j, so the information will not be mixed with other log4j logging. the `dryrun` is kind of printing an execution plan.

Reply:
Can't it be separated out using a different Logger object?

Reply:
in my opinion, the output of a command line is totally different from the logging information. they are served for different purpose. typically when users type a command with `dryrun` in bk shell, they would expect something are printed in the console to tell the result of `dryrun`.  that means in the code we anyway need to distinguish output of a command and the logging information. using a different logger object doesn't help here, because that means people need to be aware of the implementation and customize their log4j properties in order to get the right result of `dryrun` when using bk shell, that is going to be a pain point for users.

If you feel strong about `System.out`, I can wrapp this into a method call `verbose(...)`. does that work for you?

Reply:
What I'd like to avoid is that someone is using BookKeeperAdmin in some service, and they end up with strange messages on stdout. This same effect as System.out a custom logger. We provide the log4j.properties.

Create the Logger
Logger verbose = LoggerFactory.getLogger("verbose")

Then in log4j.shell.properties put
log4j.logger.verbose=INFO,VERBOSECONSOLE
log4j.appender.VERBOSECONSOLE=org.apache.log4j.ConsoleAppender
log4j.appender.VERBOSECONSOLE.Threshold=INFO
log4j.appender.VERBOSECONSOLE.layout=org.apache.log4j.PatternLayout
log4j.appender.VERBOSECONSOLE.layout.ConversionPattern=%m%n

Reply:
`dryrun` is a specified by the caller, it is a contract between the caller and the implementation. if someone is using BookKeeperAdmin in some service, they are aware of this behavior since it is a contract provided by the admin. the behavior is expected and determined when using `stdout`.

but using log4j for tool command results make the behavior is a bit undefined. because the whole behavior is tied with a log4j.properties file. it is fine if people using the bk shell, but behavior would become undefined if people integrates BookKeeperAdmin in their tools. they have to be aware of the implementation of `dryrun` and customize their log4j properties accordingly.

anyway, I will change this to log4j to move forward here and update bkshell accordingly in the subsequent change.

Reply:
For bkshell we need both console log and actual log file log. Can we have that setting only for bkshell?
For regular bk/auditor/replication worker log should go to actual log.

Reply:
@jvrao we were only talking about the output introduced by `dryrun`. we were not discussing changing the existing logging behavior.